src/System.Runtime.Extensions/src/System/Diagnostics/Stopwatch.cs

```diff
@@ -29,14 +29,15 @@ public partial class Stopwatch
         // to ticks. 
         private static readonly double s_tickFrequency;
 
-        static Stopwatch()
+        static unsafe Stopwatch()
         {
-            bool succeeded = QueryPerformanceFrequency(out Frequency);
+            long frequency;
+            int succeeded = QueryPerformanceFrequency(&frequency);
```

Comment:
The static cctor is the only place QueryPerformanceFrequency is called, right?  The minute additional overhead to having a simpler P/Invoke signature matters in that case?

Reply:
Yes, but it is fairly simple to just avoid the overhead anyways, so why not?

Reply:
Because it complicates the P/Invoke signature and call sites leading up to it and forces the code using it to be unsafe.  When it was using "out", the unsafeness could be constrained to just the P/Invoke signature, and if that was guaranteed correct, the callers would also be safe; that's no longer the case here.

Reply:
We have been on a path towards PInvoke signatures without marshaling. I think it is fine to be flipping the PInvoke signatures to no marshaling variety as we see fit. I do not think it is worth it to distinguish hot PInvokes and cold PInvokes. It would just lead to style inconsistencies.

If you need to contain the `unsafe` because of the PInvoke is used in many places, you can always include a small wrapper inside the `Interop.*` file. Many `Interop.*` files include wrappers for variety of reasons already.

Reply:
> I do not think it is worth it to distinguish hot PInvokes and cold PInvokes. It would just lead to style inconsistencies.

Fair enough.